core/group_setup.go

```diff
@@ -266,7 +266,7 @@ func (s *setupManager) createAndSend(keys []*key.Identity) {
 		_, transition := chain.NextRound(atLeast, s.beaconPeriod, s.oldGroup.GenesisTime)
 
 		group = key.NewGroup(keys, s.thr, genesis, s.beaconPeriod, s.catchupPeriod, s.scheme, s.beaconID)
-		group.TransitionTime = transition
+		group.TransitionTime = transition + int64(s.beaconPeriod.Seconds())
```

Comment:
I think this is handled above on line 262 by the `atLeast` variable where we do  `.Add(s.beaconPeriod)`

Reply:
Good catch! I've removed this.